src/impl/wshandshake.cpp

```diff
@@ -73,6 +78,11 @@ string WsHandshake::generateHttpRequest() {
 	if (!mProtocols.empty())
 		out += "Sec-WebSocket-Protocol: " + utils::implode(mProtocols, ',') + "\r\n";
 
+	// Add custom headers
+	for (const auto& [headerName, headerValue] : mCustomHeaders) {
+		out += headerName + ": " + headerValue + "\r\n";
+	}
```

Comment:
I think you should at least sanitize `headerValue`:
```suggestion
	for (auto [headerName, headerValue] : mCustomHeaders) {
	    headerValue.erase(std::remove(headerValue.begin(), headerValue.end(), '\r'), headerValue.end());
	    std::replace(headerValue.begin(), headerValue.end(), '\n', ' ');
		out += headerName + ": " + headerValue + "\r\n";
	}
```